--- origsrc/svn/libskype.c	2008-05-20 07:57:18.000000000 -0500
+++ src/svn/libskype.c	2008-05-20 08:40:21.187500000 -0500
@@ -135,7 +135,7 @@
 
 PurplePluginProtocolInfo prpl_info = {
 	/* options */
-#if _WIN32 || __APPLE__ || SKYPE_DBUS || !USE_XVFB_SERVER
+#if _WIN32 || __CYGWIN__ || __APPLE__ || SKYPE_DBUS || !USE_XVFB_SERVER
 	OPT_PROTO_NO_PASSWORD|
 #endif
 	OPT_PROTO_REGISTER_NOSCREENNAME|OPT_PROTO_CHAT_TOPIC|OPT_PROTO_SLASH_COMMANDS_NATIVE|OPT_PROTO_UNIQUE_CHATNAME,
@@ -578,7 +578,7 @@
 	gchar version_url[60];
 	int platform_number;
 
-#ifdef _WIN32
+#if defined(_WIN32) || defined(__CYGWIN__)
 	platform_number = 0;
 #else
 #	ifdef __APPLE__
--- origsrc/svn/skype_messaging.c	2008-05-20 07:57:18.000000000 -0500
+++ src/svn/skype_messaging.c	2008-05-20 08:28:20.234375000 -0500
@@ -19,6 +19,10 @@
 #ifdef _WIN32
 #	include "skype_messaging_win32.c"
 #else /*if !win32 */
+#   ifdef __CYGWIN__
+#	    include "skype_messaging_win32.c"
+#	    undef _WIN32
+#   else
 #	ifdef __APPLE__
 #		include "skype_messaging_carbon.c"
 #	else /*if !apple */
@@ -27,6 +31,7 @@
 #		else 
 #			include "skype_messaging_dbus.c"
 #		endif /* !x11 */
+#	endif /* !cygwin */
 #	endif /* !apple */
 #endif /* win32 */
 
--- origsrc/svn/skype_messaging_win32.c	2008-05-20 07:57:18.000000000 -0500
+++ src/svn/skype_messaging_win32.c	2008-05-20 08:15:55.968750000 -0500
@@ -1,6 +1,6 @@
+#include <glib.h>
 #include <windows.h>
 #include <tlhelp32.h>
-#include <glib.h>
 
 #define SKYPE_WIN32_CLASS_NAME "Skype-libpurple-Joiner"
 
